script.js

```diff
@@ -0,0 +1,160 @@
+// @ts-check
+
+import { composeCreateOrUpdateTextFile } from "@octokit/plugin-create-or-update-text-file";
+import prettier from "prettier";
+import YAML from 'yaml'
+
+const { parseDocument, parse, stringify, Document } = YAML
+
+const isYamlFile = (fileName) => {
+  const yamlFilePattern = /^.*\.ya?ml$/;
+  return yamlFilePattern.test(fileName);
+};
+
+/**
+ * Add cache parameter to GitHub Actions using setup-node
+ *
+ * @param {import('@octoherd/cli').Octokit} octokit
+ * @param {import('@octoherd/cli').Repository} repository
+ * @param {object} options
+ * @param {string} [options.cache] Select which package manager you want to use for caching
+ */
+export async function script(octokit, repository, { cache = "npm" }) {
+  // Global variables used throughout the code
+  const owner = repository.owner.login;
+  const repo = repository.name;
+  const defaultBranch = repository.default_branch;
+  const branchName = "add-cache-to-node-workflows";
+  const path = ".github/workflows";
```

Comment:
maybe move these outside of the script and capitalize their names to make clear these are global constants?